### PR TITLE
feat(core): compile only schematics realted source when using workspa…

### LINKED ISSUE #1215: Exclude certain sub directories from tools from being copied when using custom schematics
Hi, I'm using the tools directory not just for schematics, but also for some other tooling (mainly some Cake files for other build steps). Currently the schematics tooling is copying all directories and files in the tools directory to the output directory and because of that, from time to time I'm seeing some errors complaining the outdir is not empty, not exactly sure when that happens.

I'd like to have the options to exclude some directories (maybe by using an "exclude" section in `tsconfig.tools.json`?). So when I have this directory:
```
/tools
	/Addins
	/Cake
	/schematics
	tsconfig.tools.json	
```

I would like to see only the schematics folder being copied to the outDir.

This is the current implementation in https://github.com/nrwl/nx/blob/master/packages/schematics/src/command-line/workspace-schematic.ts 

```ts
function compileToolsDir(outDir: string) {
  copySync(path.join(rootDirectory, 'tools'), outDir);
  try {
    execSync('tsc -p tools/tsconfig.tools.json', {
      stdio: 'inherit',
      cwd: rootDirectory
    });
  } catch (e) {
    process.exit(1);
  }
}
```

### PR DESCRIPTION
**Pull request by Hotell**
…ce-schematics

- this also improves  performance when invoking `nx workspace-schematic` (because less files need to be compiled)
- we are generating temp tsconfig, as using `exclude` capabilities as proposed in original issue would introduce bugs if one uses files outside `/schematics` folder ( our usecase at work for instance )

ISSUES CLOSED: #1215

## Related Issue(s)

Fixes #

### CODE REVIEW

**Comment by FrozenPandaz** [comment]
Cool!

Question: If somebody imports from say `tools/utils`, that would not be in the compilation and it would break right?

There might be a way to optimize this by only including `tools/schematics/**/index.ts` instead of `**/*.ts`?

**Comment by Hotell** [comment]
> Question: If somebody imports from say tools/utils, that would not be in the compilation and it would break right?

- it would not break, why should it (thats wrong behaviour) ?

For instance we have following `tools/` structure in our project:

image

So `tools/utils` need to be compiled as it's a dependency for schematics ( without schematics, would not work)

> There might be a way to optimize this by only including tools/schematics/**/index.ts instead of **/*.ts?

I don't think it's good idea, nor will it speed up radically:

1. any file imported in that `index.ts` would be compiled anyway 

2. whilst workspace schematic generates 

```ts
/schematics/
    - /schematic-name
              - index.ts
              - schema.json
```

Anyone can change that index.ts to other name. ( after my existing PR for workspace-schematics will be merged hopefully ), I wanna submit another one that will support proper testing and static collection generation, which will remove need of dynamically generating collection on every run, thus will allow users to name their schematics file implementation as they want )

**Different approach for the future:**

I'd recommend to follow standard libraries hierarchy for `tools` (we are doing this manually anyway and I saw other teams/companies to do that as well). 
That means  having `tsconfig.json` and `tsconfig.spec.json`,`tsconfig.lib.json`, and then narrowing compilation for particular supported feature like presented in this PR.

File: e2e/workspace/src/workspace-aux-commands.test.ts [hunk 1]
```diff
@@ -117,7 +118,9 @@ forEachCli((cli) => {
         );
       });
     });
+  });
 
+  describe('format', () => {
```
Comment (Hotell) [inline e2e/workspace/src/workspace-aux-commands.test.ts:123]: I've grouped relevant commands under common `describe` block

**Comment by vsavkin** [comment]
Thank you!

**Pull request merged**